Tools/Eflow/provision-eflow.ps1

```diff
@@ -1,109 +1,85 @@
 <# .SYNOPSIS
     EFLOW Windows Server provisioning Script
 .DESCRIPTION
-    THIS SCRIPT IS NOT RECOMMENDED TO BE USED AS IS, BUT SHOULD BE SEEN AS A COLLECTION OF STEPS TO BE RUN MANUALLY
-    TO GET EFLOW UP AND RUNNING.
-    This script automatically install Edge For Linux On Windows (EFLOW) on a new Windows Server VM
-    The script might trigger a restart and might need to be rerun after a restart
+    This script automatically installs Edge For Linux On Windows (EFLOW) on a new Windows Server VM.
+    This script assumes that the Windows VM has Hyper-V already installed.
+    The script accepts the following parameters:
```

Comment:
Let's double check about internet connection sharing with the EFLOW team. If this is needed, we should add an item in the backlog to eventually improve this script for automating such connection sharing.

Reply:
I tried to automate this, and it _kind of_ worked. 

The powershell script I used (See [here](https://superuser.com/questions/470319/how-to-enable-internet-connection-sharing-using-command-line))
```
regsvr32 hnetcfg.dll
$m = New-Object -ComObject HNetCfg.HNetShare
$c = $m.EnumEveryConnection |? { $m.NetConnectionProps.Invoke($_).Name -eq "Ethernet" }
$config = $m.INetSharingConfigurationForINetConnection.Invoke($c)
$config.EnableSharing(0)
```

After running this, if you check the network connections/properties/sharing check box, you will see that it is checked. However, the desired effect (a change in the IP address of the VM switch) doesn't happen. I could only make it happen when checking the box manually.  

I have to stop working on this for now, that's why I documented the manual approach. But I can open an issue describing what I tried for whoever tries to implement this next.

Reply:
Here is the issue where I found the idea : https://github.com/Azure/iotedge-eflow/issues/59#issuecomment-893620909

Reply:
I think the issue is related to the fact that we are creating an "Internal" virtual switch.
Can we try using an ["External" virtual switch](https://learn.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows-networking?view=iotedge-1.4#virtual-switch-choices) instead? I think we did not use it at the beginning as it was not supported for EFLOW, but now seems to be supported for both windows server and client environments.

Reply:
The [doc](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-create-virtual-switch?view=iotedge-1.4#create-virtual-switch) seems to indicate you can't. But we will try anyway

Reply:
I've also tried it
![image](https://user-images.githubusercontent.com/18661075/199444521-073a9ec9-2f2c-4b05-a14e-29a35bc7e8e2.png)

Reply:
See docs [here](https://learn.microsoft.com/en-us/powershell/module/hyper-v/new-vmswitch?view=windowsserver2022-ps#-switchtype) on how to create an external switch.

@Mandur, I just saw the note, but we could give it a try anyways

Reply:
I tried again. The switch is assigned an internal IP, and EFLOW does not have access to the internet. 
![image](https://user-images.githubusercontent.com/18661075/199465299-fe3986cb-e6e8-4ed6-bb21-ab620365bec5.png)

![image](https://user-images.githubusercontent.com/18661075/199465851-ea04d7d9-f430-4964-a59a-a929b7603267.png)

Reply:
Ok, let's keep as it is. Will maybe investigate in the future 👍